frontend/apps/reader/modules/readerlink.lua

```diff
@@ -1405,4 +1399,32 @@ function ReaderLink:showAsFootnotePopup(link, neglect_current_location)
     return true
 end
 
+function ReaderLink:addButton(idx, fn_button)
+    self.alt_handlers_buttons[idx] = fn_button
+end
+
+function ReaderLink:removeButton(idx)
```

Comment:
I think I mentionned that in your previous PR (or one of them :), but you'd better try to keep the same interface/API as ReaderHighlight, which already allows plugins to add buttons (although none currently use it).

Curious: which of your plugins will add what buttons ?

Reply:
``` lua
function ReaderHighlight:addToHighlightDialog(idx, fn_button)
    -- fn_button is a function that takes the ReaderHighlight instance
    -- as argument, and returns a table describing the button to be added.
    self._highlight_buttons[idx] = fn_button
end
```

It's fairly similar. I can def change the method name to be something like `addToExternalLinkDialog`.

I have a new plugin that will download a link to an EPUB. That would add a single button to "Download to EPUB". (Often, I'm reading articles from the RSS reader, and they contain links, and I want to read the links! So hence the "download to EPUB" :)  )